### PR TITLE
Refactored Code

### PR DESCRIPTION
**Pull request by visheshjindal**
* Updated MainViewModel
* Did minor refactoring across the code
* Minor UI Update

### CODE REVIEW

File: app/src/main/java/com/example/androidreduxsample/redux/Store.kt [hunk 1]
```diff
@@ -20,15 +25,15 @@ class Store @Inject constructor(
         )
     )
 
-    val appStateFlow: StateFlow<AppState> = _appStateFlow.asStateFlow()
+    val appStateFlow: StateFlow<AppState>
+        get() = _appStateFlow.asStateFlow()
```
Comment (anandpc) [inline app/src/main/java/com/example/androidreduxsample/redux/Store.kt:29]: does this makes any difference from the existing ?

File: app/src/main/java/com/example/androidreduxsample/ui/viewmodel/MainViewModel.kt [hunk 2]
```diff
@@ -4,38 +4,50 @@ import androidx.lifecycle.ViewModel
...
-    val articles: StateFlow<List<Article>> = _articles.asStateFlow()
+    /**
+     * mainScreenState is a [StateFlow] that emits the current and new state updates of [MainScreenState]
+     * [StateFlow] is a state-holder observable flow that emits the current and new state updates
+     */
+    val mainScreenState: StateFlow<MainScreenState> = store.appStateFlow.map { appState ->
+        appState.mainScreenState
+    }.stateIn(
+        scope = viewModelScope,
+        started = SharingStarted.WhileSubscribed(stopTimeoutMillis = 5000),
```
Comment (anandpc) [inline app/src/main/java/com/example/androidreduxsample/ui/viewmodel/MainViewModel.kt:39]: please can you explain why 5000 only ?

File: app/src/main/java/com/example/androidreduxsample/ui/screen/MainScreen.kt [hunk 3]
```diff
@@ -76,30 +101,27 @@ fun ArticleList(articles: List<Article>) {
 
 @Composable
 fun ArticleItem(article: Article, onArticleClick: (Int) -> Unit) {
+    val imagePainter = rememberImagePainter(article.urlToImage)
```
Comment (anandpc) [inline app/src/main/java/com/example/androidreduxsample/ui/screen/MainScreen.kt:104]: we lifted this up at func() level, any reason ?

File: app/src/main/java/com/example/androidreduxsample/redux/Store.kt (hunk 1 above)
Comment (visheshjindal) [reply app/src/main/java/com/example/androidreduxsample/redux/Store.kt:29]: its a computed property, means I am not storing it in variable but getting it on-the-fly.

File: app/src/main/java/com/example/androidreduxsample/ui/viewmodel/MainViewModel.kt (hunk 2 above)
Comment (visheshjindal) [reply app/src/main/java/com/example/androidreduxsample/ui/viewmodel/MainViewModel.kt:39]: If app goes in background. I don't believe I need to listen to the stream further. So, I kept standard 5000. Can remove it as well.

File: app/src/main/java/com/example/androidreduxsample/ui/screen/MainScreen.kt (hunk 3 above)
Comment (visheshjindal) [reply app/src/main/java/com/example/androidreduxsample/ui/screen/MainScreen.kt:104]: Better readability

**Pull request merged**